src/apps/tap/tap.lua

```diff
@@ -68,7 +67,9 @@ function Tap:pull ()
       if ethernet:is_bcast(p.data) then
          counter.add(self.shm.rxbcast)
       end
+      p = packet.allocate()
    end
+   self.pkt = p
```

Comment:
Doesn't that result in an extra copy for every packet?
I guess worrying about extra copies is probably mute given that it's a Tap interface, but it seems a slight dubious habit to get into.